### PR TITLE
Add InterRDF_s

### LINKED ISSUE #60: A parallel version of InterRDF_s
Current version of pmda.rdf doesn't contain a parallelled mdanalysis.analysis.rdf.InterRDF_s, which is a tool to calculate site-specific radial distribution function. We need to add this to the current pmda.rdf.

### PR DESCRIPTION
**Pull request by VOD555**
Fixes #60

Changes made in this Pull Request:
 - Add parallel version of InterRDF_s(in pmda.rdf)
 - Add test for InterRDF_s

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: pmda/rdf.py [hunk 1]
```diff
@@ -172,3 +172,181 @@ def _reduce(res, result_single_frame):
...
+                                                    self._maxrange,
+                                                    box=u.dimensions)
+
+            for j, (idx1, idx2) in enumerate(pairs):
+                count[i][idx1, idx2, :] = np.histogram(dist[j],
+                                                       **self.rdf_settings)[0]
+
+        volume = u.trajectory.ts.volume
+
+        return np.array([np.array(count), np.array(volume, dtype=np.float64)])
```
[the remark this answers is not in the document]
Comment (VOD555) [reply pmda/rdf.py]: Yes, this return value can do numpy array addition.

File: pmda/test/test_rdf_s.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    res = rdf_s._reduce(res, single_frame)
+    assert_almost_equal(res[0], np.array([2, 4]))
+    assert_almost_equal(res[1], np.array([6]))
+
+
+def test_same_result(u, sels):
+    # should see same results from analysis.rdf.InterRDF_s
+    # and pmda.rdf.InterRDF_s
+    nrdf = rdf.InterRDF_s(u, sels).run()
+    prdf = InterRDF_s(u, sels).run(n_jobs=3)
```
Comment (VOD555) [inline pmda/test/test_rdf_s.py]: @orbeckst Here is one that is testing multiple blocks. Maybe I should add one test for "n_blocks=1" case

**Comment by VOD555** [comment]
@orbeckst An error is raised on the use of the get= keyword in the new version of dask(http://docs.dask.org/en/latest/changelog.html)

**Comment by orbeckst** [comment]
That happened quickly! 

So we now HAVE to finish PR #60!

File: pmda/test/test_rdf_s.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    assert_almost_equal(res[1], np.array([6]))
+
+
+@pytest.mark.parametrize("n_blocks", [1, 2])
+def test_same_result(u, sels, n_blocks):
+    # should see same results from analysis.rdf.InterRDF_s
+    # and pmda.rdf.InterRDF_s
+    nrdf = rdf.InterRDF_s(u, sels).run()
+    prdf = InterRDF_s(u, sels).run(n_blocks=n_blocks)
+    assert_almost_equal(nrdf.count[0][0][0], prdf.count[0][0][0])
```
Comment (orbeckst) [inline pmda/test/test_rdf_s.py]: Why compare just one element? Shouldn't
```python
assert_almost_equal(nrdf.count, prdf.count)
```
work?

File: pmda/test/test_rdf_s.py [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    s2 = u.select_atoms(
+         '(name OD1 or name OD2) and resid 51 and sphzone 5.0 (resid 289)')
+    s3 = u.select_atoms('name ZND and (resid 291 or resid 292)')
+    s4 = u.select_atoms('(name OD1 or name OD2) and sphzone 5.0 (resid 291)')
+    ags = [[s1, s2], [s3, s4]]
+    return ags
+
+
+@pytest.fixture(scope='module')
+def rdf_s(u, sels):
```
Comment (orbeckst) [inline pmda/test/test_rdf_s.py]: Test with different schedulers:
```python
def rdf_s(u, sels, scheduler)
```

Comment (orbeckst) [reply pmda/test/test_rdf_s.py]: (The `scheduler` fixture is defined in `conftest.py` and should run this calculation with all dask schedulers.)

**Comment by VOD555** [comment]
Gets error with `distributed`

```python
=========================================================================================== test session starts ============================================================================================
platform linux -- Python 3.6.5, pytest-3.6.2, py-1.5.3, pluggy-0.6.0 -- /home/shujie/anaconda3/envs/py36/bin/python
cachedir: ../../.pytest_cache
rootdir: /home/shujie/pmda, inifile: setup.cfg
plugins: pep8-1.0.6, cov-2.5.1, hypothesis-3.66.1
collected 38 items                                                                                                                                                                                         

test_rdf_s.py::test_nbins PASSED                                                                                                                                                                     [  2%]
test_rdf_s.py::test_range PASSED                                                                                                                                                                     [  5%]
test_rdf_s.py::test_count_size[distributed-1] ERROR                                                                                                                                                  [  7%]
test_rdf_s.py::test_count[distributed-1] ERROR                                                                                                                                                       [ 10%]
test_rdf_s.py::test_double_run[distributed-1] ERROR                                                                                                                                                  [ 13%]
test_rdf_s.py::test_cdf[distributed-1] ERROR                                                                                                                                                         [ 15%]
test_rdf_s.py::test_reduce[distributed-1] ERROR                                                                                                                                                      [ 18%]
test_rdf_s.py::test_count_size[multiprocessing-1] PASSED                                                                                                                                             [ 21%]
test_rdf_s.py::test_count[multiprocessing-1] PASSED                                                                                                                                                  [ 23%]
test_rdf_s.py::test_double_run[multiprocessing-1] PASSED                                                                                                                                             [ 26%]
test_rdf_s.py::test_cdf[multiprocessing-1] PASSED                                                                                                                                                    [ 28%]
test_rdf_s.py::test_reduce[multiprocessing-1] PASSED                                                                                                                                                 [ 31%]
test_rdf_s.py::test_count_size[multiprocessing-2] PASSED                                                                                                                                             [ 34%]
test_rdf_s.py::test_count[multiprocessing-2] PASSED                                                                                                                                                  [ 36%]
test_rdf_s.py::test_double_run[multiprocessing-2] PASSED                                                                                                                                             [ 39%]
test_rdf_s.py::test_cdf[multiprocessing-2] PASSED                                                                                                                                                    [ 42%]
test_rdf_s.py::test_reduce[multiprocessing-2] PASSED                                                                                                                                                 [ 44%]
test_rdf_s.py::test_count_size[distributed-2] ERROR                                                                                                                                                  [ 47%]
test_rdf_s.py::test_count[distributed-2] ERROR                                                                                                                                                       [ 50%]
test_rdf_s.py::test_double_run[distributed-2] ERROR                                                                                                                                                  [ 52%]
test_rdf_s.py::test_cdf[distributed-2] ERROR                                                                                                                                                         [ 55%]
test_rdf_s.py::test_reduce[distributed-2] ERROR                                                                                                                                                      [ 57%]
test_rdf_s.py::test_count_size[single-threaded-2] PASSED                                                                                                                                             [ 60%]
test_rdf_s.py::test_count[single-threaded-2] PASSED                                                                                                                                                  [ 63%]
test_rdf_s.py::test_double_run[single-threaded-2] PASSED                                                                                                                                             [ 65%]
test_rdf_s.py::test_cdf[single-threaded-2] PASSED                                                                                                                                                    [ 68%]
test_rdf_s.py::test_reduce[single-threaded-2] PASSED                                                                                                                                                 [ 71%]
test_rdf_s.py::test_count_size[single-threaded-1] PASSED                                                                                                                                             [ 73%]
test_rdf_s.py::test_count[single-threaded-1] PASSED                                                                                                                                                  [ 76%]
test_rdf_s.py::test_double_run[single-threaded-1] PASSED                                                                                                                                             [ 78%]
test_rdf_s.py::test_cdf[single-threaded-1] PASSED                                                                                                                                                    [ 81%]
test_rdf_s.py::test_reduce[single-threaded-1] PASSED                                                                                                                                                 [ 84%]
test_rdf_s.py::test_same_result[1] PASSED                                                                                                                                                            [ 86%]
test_rdf_s.py::test_same_result[2] PASSED                                                                                                                                                            [ 89%]
test_rdf_s.py::test_same_result[3] PASSED                                                                                                                                                            [ 92%]
test_rdf_s.py::test_same_result[4] PASSED                                                                                                                                                            [ 94%]
test_rdf_s.py::test_density[True-13275.775528444701] PASSED                                                                                                                                          [ 97%]
test_rdf_s.py::test_density[False-0.021915460340071267] PASSED                                                                                                                                       [100%]

================================================================================================== ERRORS ==================================================================================================
```

File: pmda/rdf.py [hunk 5]
```diff
@@ -172,3 +172,182 @@ def _reduce(res, result_single_frame):
...
+    def __init__(self, u, ags,
+                 nbins=75, range=(0.0, 15.0), density=True):
+        atomgroups = []
+        for pair in ags:
+            atomgroups.append(pair[0])
+            atomgroups.append(pair[1])
+        super(InterRDF_s, self).__init__(u, atomgroups)
+
+        # List of pairs of AtomGroups
+        self.ags = ags
```
Comment (VOD555) [inline pmda/rdf.py]: Ok, the failure is due to this line. `distributed` cannot pass atomgroups to blocks.

Comment (orbeckst) [reply pmda/rdf.py]: Seems related to PR #65 .

(Also, thanks for raising #79 )

**Pull request merged**